great_expectations/self_check/util.py

```diff
@@ -1571,15 +1616,63 @@ def build_sa_validator_with_data(  # noqa: C901 - 39
     )
 
     batch_data = SqlAlchemyBatchData(execution_engine=engine, table_name=table_name)
-    batch = Batch(data=batch_data, batch_definition=batch_definition)
     execution_engine = SqlAlchemyExecutionEngine(caching=caching, engine=engine)
 
+    if context is None:
+        context = build_in_memory_runtime_context()  # type: ignore[assignment]
+
+    context.datasources["my_test_datasource"] = Datasource(  # type: ignore[union-attr]
```

Comment:
Same note here and for all other `type: ignore` comments - we should use them __very__ sparingly so we don't bury type errors

Reply:
@cdkini These are not crucial for now (in tests -- all of which will change when we eliminate `instantiate_class_from_config`).

Reply:
What are the possible return values in this `Union`? I don't see `instantiate_class_from_config` going anywhere anytime soon due to how deeply entrenched it is so I'm attempting to be a bit stricter until we get there.